src/provider/provider_tracking.c

```diff
@@ -203,14 +378,16 @@ static umf_result_t trackingAllocationSplit(void *hProvider, void *ptr,
 
     splitValue->pool = provider->pool;
     splitValue->size = firstSize;
+    splitValue->n_children = 0;
 
     int r = utils_mutex_lock(&provider->hTracker->splitMergeMutex);
     if (r) {
         goto err_lock;
     }
 
-    tracker_alloc_info_t *value = (tracker_alloc_info_t *)critnib_get(
-        provider->hTracker->alloc_segments_map, (uintptr_t)ptr);
+    int level = 0;
+    tracker_alloc_info_t *value = critnib_get_last_no_children(
```

Comment:
explain in the comment why "no children"

Reply:
Done